src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -220,6 +341,18 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     throw;
                 }
             }
+            finally
+            {
+                if (innerContext != null)
+                {
+                    // We allocated new context and we want to cache
```

Comment:
This comment doesn't match the code.  The block it's attached to is that we don't want to cache.